src/EventListener/PrettyErrorScreenListener.php

```diff
@@ -0,0 +1,312 @@
+        $this->twig               = $twig;
+        $this->config             = $config;
+    }
+
+    /**
+     * Map the exception to an error screen.
+     *
+     * @param GetResponseForExceptionEvent $event The event object
+     *
+     * @todo Log the exception?
+     */
+    public function onKernelException(GetResponseForExceptionEvent $event)
+    {
+        $exception = $event->getException();
+
+        switch (true) {
+            case $exception instanceof AccessDeniedHttpException:
+                $this->renderErrorScreen(403, $event);
+                break;
+
+            case $exception instanceof BadRequestHttpException:
+            case $exception instanceof InternalServerErrorHttpException:
+                $this->checkExceptionChain($event);
+                break;
+
+            case $exception instanceof NotFoundHttpException:
+                $this->renderErrorScreen(404, $event);
+                break;
+
+            case $exception instanceof ServiceUnavailableHttpException:
+                $this->renderMaintenanceScreen($event);
+                break;
+        }
+    }
+
+    /**
+     * Renders the error screen.
+     *
+     * @param GetResponseForExceptionEvent $event The event object
+     */
+    private function renderErrorScreen($type, GetResponseForExceptionEvent $event)
+    {
+        static $processing;
+
+        if (true === $processing) {
+            return;
+        }
+
+        $processing = true;
+
+        if (null !== ($response = $this->renderPageHandler($type, $event))) {
+            $event->setResponse($response);
+        } else {
+            $this->checkExceptionChain($event);
+        }
+
+        $processing  = false;
+    }
+
+    /**
+     * Renders a Contao page handler.
+     *
+     * @param string $type The error type
+     *
+     * @return Response|null The response object or null
+     */
+    private function renderPageHandler($type)
+    {
+        $type = "error_$type";
+
+        if (!isset($GLOBALS['TL_PTY'][$type]) || !class_exists($GLOBALS['TL_PTY'][$type])) {
+            return null;
+        }
+
+        /** @var \PageError403 $pageHandler */
+        $pageHandler = new $GLOBALS['TL_PTY'][$type]();
+
+        try {
+            return $pageHandler->getResponse(false); // FIXME: the class requires a numeric $pageId
+        } catch (\Exception $e) {
+            return null;
+        }
+    }
+
+    /**
+     * Renders the maintenance screen.
+     *
+     * @param GetResponseForExceptionEvent $event The event object
+     */
+    private function renderMaintenanceScreen(GetResponseForExceptionEvent $event)
+    {
+        $event->setResponse($this->renderTemplate('service_unavailable', 503));
+    }
+
+    /**
+     * Checks the exception chain for a known exception.
+     *
+     * @param GetResponseForExceptionEvent $event The event object
+     */
+    private function checkExceptionChain(GetResponseForExceptionEvent $event)
```

Comment:
If we change the name, it should begin with `render` like the other methods, shouldn't it?